theano/tensor/opt.py

```diff
@@ -3421,7 +3423,21 @@ def local_fill_sink(node):
     c = node.op(*inputs)
     for model in models:
         c = T.fill(model, c)
-    return [c]
+
+    # The newly created node c doesn't has 'clients',
+    # so this iteration is took place with node.outputs[0]
+    replacements = {node.outputs[0]: c}
+    for client, _ in node.outputs[0].clients:
+        if (hasattr(client, 'op') and
+                isinstance(client.op, T.Elemwise) and
+                not client.op == T.fill):
+            # import ipdb; ipdb.set_trace()
+            r = local_fill_sink.transform(client)
+            if r is False:
```

Comment:
`if not r:` is more robust in general, because optimizations sometimes return `None` or `[]` as well as `False`.